site/docs/types/compound_logical_types.md

```diff
@@ -1,15 +1,15 @@
-| VARCHAR&lt;L&gt;         | A field that can holds UTF-8-encoded strings between 0 and L in character length. The value of L can be between [1..2,147,483,647]. Values shorter than L are not padded. | -                   | -              | VarcharType(L) | VARCHAR(L)                  |
-| FIXEDBINARY&lt;L&gt;     | A binary field that is fixed in width to L bytes. Values that are shorter than L are 0-padded. | FixedSizeBinary&lt;L&gt;  | FIXED&lt;L&gt;       | -              | -                           |
-| DECIMAL&lt;P, S&gt;   | A fixed-precision decimal value having precision (P, number of digits) <= 38 and scale (S, number of fractional digits) 0 <= S <= P. | Decimal&lt;P, S, bitwidth=128&gt; | DECIMAL(P,S)   | DECIMAL(P,S)   | DECIMAL(P,S)                |
-| STRUCT&lt;T1,...,T2&gt; | A list of types in a defined order. | struct_&lt;*&gt;                  | struct&lt;*&gt; | struct&lt;*&gt; | row&lt;*&gt;   |
-| NSTRUCT&lt;N:T1,...,N:T2&gt; | **Pseudo-type**: A struct that maps unique names to value types. Each name is a UTF-8-encoded string. Each value can have a distinct type. Note that NSTRUCT is actually a pseudo-type. This is because Substrait's core type system is based entirely on ordinal positions, not named fields. Nonetheless, when working with systems outside Substrait, names are important. | struct_&lt;*&gt; | struct&lt;*&gt; | struct&lt;*&gt; | row&lt;*&gt; |
-| LIST&lt;T&gt;               | A list of values of type T. The list can be between [0..2,147,483,647] values in length. | list                | list           | list           | array                       |
-| MAP&lt;K, V&gt;                   | An unordered list of type K keys with type V values.         | map&lt;k,v&gt;            | map&lt;k,v&gt;       | -              | map&lt;k,v&gt;                    |
+| Type Name                    | Description                                                  | Canonical physical representation (protobuf)
+| ---------------------------- | ------------------------------------------------------------ | ------------------------------------------------
+| FIXEDCHAR&lt;L&gt;           | A fixed-length unicode string of L characters. L must be within [1..2,147,483,647]. When casting, strings shorter than L are padded with spaces, and strings longer than L are right-trimmed. | L-character `string`
+| VARCHAR&lt;L&gt;             | A unicode string of at most L characters.L must be within [1..2,147,483,647]. When casting, strings shorter than L are not padded, and strings larger than L are right-trimmed. | `string` with at most L characters
+| FIXEDBINARY&lt;L&gt;         | A binary string of L bytes. When casting, values shorter than L are padded with zeros, and values longer than L are right-trimmed. | L-byte `bytes`
```

Comment:
I agree, but the original text does define what I can only describe as casting behavior there. I left it there since casts are not otherwise defined yet.